src/gui/transferlistfilterswidget.cpp

```diff
@@ -428,12 +428,12 @@ void TrackerFiltersList::addItem(const QString &tracker, const BitTorrent::Torre
 
     if (exists)
     {
-        if (m_trackers.value(host).contains(id))
+        if (m_trackers.value(host).torrents.contains(id))
             return;
 
-        trackerItem = item((host == NULL_HOST)
-            ? TRACKERLESS_ROW
-            : rowFromTracker(host));
+        trackerItem = (host == NULL_HOST)
+                ? item(TRACKERLESS_ROW)
```

Comment:
Can't it be stored in `m_trackers` too?

Reply:
these are treated differently everywhere and I would not change the existing structure of code. That's out of scope of this PR.